lib/runtime/procs/read.js

```diff
@@ -126,6 +126,12 @@ class Read extends source {
             this.logger.debug('scheduling first live read at', firstRead.toJSON());
             this.program.scheduler.schedule(firstRead.unixms(), _.bind(this.run, this));
         }
+
+        if (this.params.optimization_info._reduce_every && this.to.isEnd()) {
```

Comment:
I'd still like a check that throws an error if the user explicitly passed an `-every` option to read.